embassy-lora/src/stm32wl.rs

```diff
@@ -0,0 +1,87 @@
+use core::fmt::Debug;
+
+use embassy_stm32::peripherals::SUBGHZSPI;
+use embassy_stm32::{pac, spi as stm_spi, Peripheral};
+use embedded_hal_async::spi::{self, Operation};
+
+pub struct SubGhzSpiDevice<'d, Tx, Rx>
+where
+    Tx: stm_spi::TxDma<SUBGHZSPI>,
+    Rx: stm_spi::RxDma<SUBGHZSPI>,
+{
+    bus: stm_spi::Spi<'d, SUBGHZSPI, Tx, Rx>,
+}
+
+impl<'d, Tx, Rx> SubGhzSpiDevice<'d, Tx, Rx>
+where
+    Tx: stm_spi::TxDma<SUBGHZSPI>,
+    Rx: stm_spi::RxDma<SUBGHZSPI>,
+{
+    pub fn new(
+        spi: impl Peripheral<P = SUBGHZSPI> + 'd,
+        txdma: impl Peripheral<P = Tx> + 'd,
+        rxdma: impl Peripheral<P = Rx> + 'd,
+    ) -> Self {
+        let bus = stm_spi::Spi::new_subghz(spi, txdma, rxdma);
+        Self { bus }
+    }
+}
+
+/// Error returned by SPI device implementations in this crate.
+#[derive(Eq, PartialEq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+pub enum SpiDeviceError {
+    /// An operation on the inner SPI bus failed.
+    Spi(stm_spi::Error),
+    /// DelayUs operations are not supported when the `time` Cargo feature is not enabled.
+    DelayUsNotSupported,
+}
+
+impl spi::Error for SpiDeviceError {
+    fn kind(&self) -> spi::ErrorKind {
+        match self {
+            Self::Spi(e) => e.kind(),
+            Self::DelayUsNotSupported => spi::ErrorKind::Other,
+        }
+    }
+}
+
+impl<Tx, Rx> spi::ErrorType for SubGhzSpiDevice<'_, Tx, Rx>
+where
+    Tx: stm_spi::TxDma<SUBGHZSPI>,
+    Rx: stm_spi::RxDma<SUBGHZSPI>,
+{
+    type Error = SpiDeviceError;
+}
+
+impl<Tx, Rx> spi::SpiDevice for SubGhzSpiDevice<'_, Tx, Rx>
+where
+    Tx: stm_spi::TxDma<SUBGHZSPI>,
+    Rx: stm_spi::RxDma<SUBGHZSPI>,
+{
+    async fn transaction(&mut self, operations: &mut [Operation<'_, u8>]) -> Result<(), Self::Error> {
+        pac::PWR.subghzspicr().modify(|w| w.set_nss(false));
+
+        let op_res: Result<(), stm_spi::Error> = try {
+            for op in operations {
+                match op {
+                    Operation::Read(buf) => self.bus.read(buf).await?,
+                    Operation::Write(buf) => self.bus.write(buf).await?,
+                    Operation::Transfer(read, write) => self.bus.transfer(read, write).await?,
+                    Operation::TransferInPlace(buf) => self.bus.transfer_in_place(buf).await?,
+                    #[cfg(not(feature = "time"))]
+                    Operation::DelayUs(_) => return Err(SpiDeviceError::DelayUsNotSupported),
```

Comment:
try blocks catch `?` but not `return`. so this will return without deasserting NSS.

Reply:
oh, oops. yes please!  :D

Reply:
I took a stab at this, it works but I feel like there should be a less verbose/cleaner way of doing it.